zerver/lib/markdown/__init__.py

```diff
@@ -473,75 +475,45 @@ def fetch_tweet_data(tweet_id: str) -> Optional[Dict[str, Any]]:
     return res
 
 
-HEAD_START_RE = re.compile("^head[ >]")
-HEAD_END_RE = re.compile("^/head[ >]")
-META_START_RE = re.compile("^meta[ >]")
-META_END_RE = re.compile("^/meta[ >]")
-
-
 def fetch_open_graph_image(url: str) -> Optional[Dict[str, Any]]:
-    in_head = False
-    # HTML will auto close meta tags, when we start the next tag add
-    # a closing tag if it has not been closed yet.
-    last_closed = True
-    head = []
-    # TODO: What if response content is huge? Should we get headers first?
+    og = {"image": None, "title": None, "desc": None}
+
     try:
-        content = requests.get(url, timeout=1).text
+        with requests.get(
+            url, headers={"Accept": "text/html,application/xhtml+xml"}, stream=True, timeout=1
```

Comment:
How is the timeout selected?

Reply:
The same way as before.

I didn't want to change it in this commit, especially since there isn't a good answer. 1s is both too short (e.g. not enough to reliably load zulip.com, although there are other factors at play there...) and too long (synchronously blocking Markdown processing).

Reply:
Yeah, I'll merge this and we can rethink this later; probably ultimately we want to move this preview behavior into the post-processing system and then not need to consider this question specifically.